scapy/arch/bpf/core.py

```diff
@@ -156,24 +156,26 @@ def get_working_ifaces():
         if ifflags & 0x1:  # IFF_UP
 
             # Get a BPF handle
-            fd, _ = get_dev_bpf()
+            fd = get_dev_bpf()[0]
             if fd is None:
                 raise Scapy_Exception("No /dev/bpf are available !")
 
             # Check if the interface can be used
             try:
-                fcntl.ioctl(fd, BIOCSETIF, struct.pack("16s16x", ifname.encode()))  # noqa: E501
-                interfaces.append((ifname, int(ifname[-1])))
+                fcntl.ioctl(fd, BIOCSETIF, struct.pack("16s16x",
+                                                       ifname.encode()))
+                ifnum = re.search('[0-9]*$', ifname).group(0)
```

Comment:
This will traceback when checking for `epair0a`. It needs to be something more like this:

```
matches = re.match(r"(.+?)(\d+)([ab]?)", ifname)
if matches is not None:
    interfaces.append((ifname, (matches.group(1), int(matches.group(2)), matches.group(3))
```

Example:
```
>>> r = re.compile(r"(.+?)(\d+)([ab]?)")
>>> interfaces = []
>>> for interface in ('igb0', 'em0', 'msk0', 'epair0a', 'igb1', 'vlan20', 'igb10'):
...     matches = r.match(interface)
...     if matches:
...         interfaces.append((interface, (matches.group(1), int(matches.group(2)), matches.group(3), ))
... 
>>> [interface for _, interface in sorted(interfaces, key=lambda v: v[1])]
['em0', 'epair0a', 'igb0', 'igb1', 'igb10', 'msk0', 'vlan20']
```